EIPS/eip-5593.md

```diff
@@ -37,9 +37,9 @@ The provider objects, e.g. `window.ethereum`, are expected to only inject the Et
     - The User Agent implementation MAY also support configured [potentially trustworthy origins] that would normally not be considered trustworthy if the user configures their User Agent to do so. See section 7.2 titled "Development Environments" of W3C's Secure Contexts specification for additional details. For example, in Chromium based User Agents this is done via the `chrome://flags/#unsafely-treat-insecure-origin-as-secure` flag.
 - By default the Ethereum Provider APIs MUST NOT be exposed to third-party iframes.
 - `window.ethereum` MUST be `undefined` in an iframe where `window.isSecureContext` returns `false` in that iframe.
-- If the iframe is a third party to the top-level secure origin, it SHOULD be blocked.
+- If the iframe is a third party to the top-level secure origin, it MUST be blocked.
 - If the iframe is first-party to the top-level origin AND the `sandbox` attribute is set on the iframe, the provider object MUST be blocked. If the `sandbox` attribute is set to `sandbox="allow-same-origin"` it MUST be injected for a first party frame.
-    - Note `"allow-same-origin"` does nothing if the iframe is third-party. The case of the third party iframe is dictated by the usage of the `allow` attribute and the Permissions API as defined in the rule above.
+    - Note `"allow-same-origin"` does nothing if the iframe is third-party. Third-party iframes MUST be blocked regardless of the `sandbox` attribute or any other configuration.
```

Comment:
This would make Brave's implementation non-compliant which the whole point of the EIP was to document this sandbox behavior.

The page has the ability to request usage of this via the attribute in Brave and then the user can grant or deny the permission using the permissions prompt.

I'm not sure how other wallets would handle this though.